src/js/Tables/BaseTable.js

```diff
@@ -95,40 +95,44 @@ class BaseTable {
 			return;
 		}
 		// Do nothing if no filter is found for this table.
-		const filter = window.document.querySelector(`[data-o-table-filter-id="${tableId}"]`);
-		if (!filter) {
+		const filters = window.document.querySelectorAll(`[data-o-table-filter-id="${tableId}"]`);
+		if (!filters) {
 			return;
 		}
 		// Do not setup filter if markup is missing.
 		if (!this.filterContainer) {
 			console.warn(`Could not setup the filter for the table "${tableId}" as markup is missing. A filterable table must be within a div with class "o-table-container".`);
 			return;
 		}
-		// Warn if a misconfigured filter was found.
-		const filterColumn = parseInt(filter.getAttribute('data-o-table-filter-column'), 10);
-		if (isNaN(filterColumn)) {
-			console.warn(`Could not setup the filter for the table "${tableId}" as no column index was given to filter on. Add a \`data-o-table-filter-column="{columnIndex}"\` attribute to the filter.`, filter);
-			return;
-		}
-		// Apply the filter .
-		if (filter.value) {
-			this.filter(filterColumn, filter.value);
-		}
-		// Add a listener to filter the table.
-		let pendingFilterTimeout;
-		const debouncedFilterHandler = function(event) {
-			if (pendingFilterTimeout) {
-				clearTimeout(pendingFilterTimeout);
+
+		for (const filter of Array.from(filters)) {
+
+			// Warn if a misconfigured filter was found.
+			const filterColumn = parseInt(filter.getAttribute('data-o-table-filter-column'), 10);
+			if (isNaN(filterColumn)) {
+				console.warn(`Could not setup the filter for the table "${tableId}" as no column index was given to filter on. Add a \`data-o-table-filter-column="{columnIndex}"\` attribute to the filter.`, filter);
+				return;
 			}
-			pendingFilterTimeout = setTimeout(function () {
-				this.filter(filterColumn, event.target.value || '');
-				pendingFilterTimeout = null;
-			}.bind(this), 33);
-		}.bind(this);
-		filter.addEventListener('input', debouncedFilterHandler);
-		filter.addEventListener('change', debouncedFilterHandler);
-		this._listeners.push({ element: filter, debouncedFilterHandler, type: 'input' });
-		this._listeners.push({ element: filter, debouncedFilterHandler, type: 'change' });
+			// Apply the filter .
+			if (filter.value) {
+				this.filter(filterColumn, filter.value);
```

Comment:
I think the [filter is stored](https://github.com/Financial-Times/o-table/blob/af4b8f7c3cd7faa94dc90df22e06ebd476f99095/src/js/Tables/BaseTable.js#L341) so it can be reapplied if rows change with [updateRows](https://github.com/Financial-Times/o-table/blob/af4b8f7c3cd7faa94dc90df22e06ebd476f99095/src/js/Tables/BaseTable.js#L144). At a glance, I think if rows are updated only the last filter will be reapplied?

Reply:
Oh I missed this, same as chee 😂 https://github.com/Financial-Times/o-table/pull/253#issuecomment-847730343
might be related